### PR TITLE
Instructor: search: rename tick box to be more accurate #8526

### LINKED ISSUE #8526: Instructor: search: rename tick box to be more accurate
Steps: login as an instructor and go to the search page

Current:
image

Suggested: this option widens the search to questions, responses, and comments on responses. Let's rename it to `questions, responses, comments on responses`. As it is a long name, we can swap the two options so that `students` option comes first.

### PR DESCRIPTION
**Pull request by Bzdeco**
Fixes #8526

Modified appropriate tags in `searchPageInput.tag` file. Waiting for review :smiley:

### CODE REVIEW

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag [hunk 1]
```diff
@@ -32,20 +32,20 @@
...
+          <li>
+            <input id="students-check" type="checkbox"
+                   name="<%=Const.ParamsNames.SEARCH_STUDENTS%>" value="true"
+                   <c:if test="${data.searchForStudents || !data.searchCommentForResponses}">checked</c:if>>
+            <label for="students-check">
+              Students
+            </label>
+          </li>
           <li>
             <input id="comments-for-responses-check" type="checkbox"
```
Comment (tshradheya) [inline src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Let's change the `id` to what the input is for. For e.g. this `question-responses-comments-on-responses` ?

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag [hunk 2]
```diff
@@ -32,20 +32,20 @@
...
                 name="<%=Const.ParamsNames.SEARCH_COMMENTS_FOR_RESPONSES%>" value="true"
```
Comment (tshradheya) [inline src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Update the data in `SEARCH_COMMENTS_FOR_RESPONSES` from `searchcommentforresponses` to `searchquestionresponsesandcommentsonresponses` ?

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag (hunk 1 above)
Comment (Bzdeco) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Maybe something more general and shorter like `session-feedback` would be good for it?

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag (hunk 2 above)
Comment (Bzdeco) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Likewise, `SEARCH_SESSION_FEEDBACK`?

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag (hunk 1 above)
Comment (whipermr5) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: That's too general. I think `search-questions-responses-comments-check` is good.

Comment (whipermr5) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: On second thoughts, `search-session-feedback-data-check` is good.

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag (hunk 2 above)
Comment (whipermr5) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: `SEARCH_FEEDBACK_SESSIONS_DATA = "searchfeedbacksessionsdata"`

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag (hunk 1 above)
Comment (Bzdeco) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Updated to proposed value

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag (hunk 2 above)
Comment (Bzdeco) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Updated

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag [hunk 3]
```diff
@@ -32,22 +32,22 @@
...
-            <label for="comments-for-responses-check">
-              Comments for responses
-            </label>
-          </li>
           <li>
             <input id="students-check" type="checkbox"
-                name="<%=Const.ParamsNames.SEARCH_STUDENTS%>" value="true"
-                <c:if test="${data.searchForStudents || !data.searchCommentForResponses}">checked</c:if>>
+                   name="<%=Const.ParamsNames.SEARCH_STUDENTS%>" value="true"
+                   <c:if test="${data.searchForStudents || !data.searchCommentForResponses}">checked</c:if>>
```
Comment (tshradheya) [inline src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Fix these indentation errors

Comment (Bzdeco) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Done, I had `Align attributes` checked in IntelliJ and it affected the indentation

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <%@ tag description="instructorSearch.jsp - Instructor search page input" pageEncoding="UTF-8" %>
-<%@ tag import="teammates.common.util.Const"%>
-<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c"%>
+<%@ tag import="teammates.common.util.Const" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
```
Comment (tshradheya) [inline src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Revert these unnecessary changes

File: src/main/java/teammates/ui/controller/InstructorSearchPageAction.java [hunk 5]
```diff
@@ -32,8 +32,8 @@ protected ActionResult execute() {
             numberOfSearchOptions++;
         }
 
-        boolean isSearchCommentForResponses = getRequestParamAsBoolean(Const.ParamsNames.SEARCH_COMMENTS_FOR_RESPONSES);
-        if (isSearchCommentForResponses) {
+        boolean isSearchFeedbackSessionData = getRequestParamAsBoolean(Const.ParamsNames.SEARCH_FEEDBACK_SESSION_DATA);
```
Comment (tshradheya) [inline src/main/java/teammates/ui/controller/InstructorSearchPageAction.java]: Since we have changed this, it also makes sense to rename the variables in `InstructorSearchPageData.java`

Here:
https://github.com/TEAMMATES/teammates/blob/a88f169d8b34e1e50317ac91306aa015bc68b1ba/src/main/java/teammates/ui/pagedata/InstructorSearchPageData.java#L37-L41

File: src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag (hunk 4 above)
Comment (Bzdeco) [reply src/main/webapp/WEB-INF/tags/instructor/search/searchPageInput.tag]: Done, however there is some inconsistency in it throught this and other JSP files - some tags have spaces at the end, some do not.

**Comment by Bzdeco** [comment]
Global refactoring of relevant JSP tags and classes done, hopefully I didn't miss anything that should have been changed.

File: src/main/java/teammates/ui/pagedata/InstructorSearchPageData.java [hunk 6]
```diff
@@ -55,49 +55,49 @@ public void init(FeedbackResponseCommentSearchResultBundle frcSearchResultBundle
 
         this.searchKey = searchKey;
 
-        this.isSearchCommentForResponses = isSearchCommentForResponses;
+        this.isSearchFeedbackSessionData = isSearchCommentForResponses;
```
Comment (tshradheya) [inline src/main/java/teammates/ui/pagedata/InstructorSearchPageData.java]: Change the name of argument passed to `init()` function

`this.isSearchFeedbackSessionData = isSearchCommentForResponses;` -> `this.isSearchFeedbackSessionData = isSearchFeedbackSessionData;`

**Comment by tshradheya** [comment]
While checking out this PR locally, when searching for anything it would give no results. 
I am not sure if that is due to #8662 or something broke when refactoring in this PR. @whipermr5  ?

File: src/main/java/teammates/ui/pagedata/InstructorSearchPageData.java (hunk 6 above)
Comment (Bzdeco) [reply src/main/java/teammates/ui/pagedata/InstructorSearchPageData.java]: Done

**Comment by whipermr5** [comment]
@tshradheya The search is working fine; you can search for one of the test student's names (e.g. `Alice`).

**Comment by Bzdeco** [comment]
All 3 files in help updated. While browsing the help page I also noticed that since the profile part the page was oddly rendered:
screenshot from 2018-03-24 13-41-14
The last commit seems to fix this issue - I know it was not part of this PR, but since I was editing this page I thought it should be fixed.

File: src/main/webapp/partials/instructorHelpSessions.jsp [hunk 7]
```diff
@@ -566,10 +566,7 @@
             </form>
             <br>
             <br>
-          </div>
-
         </div>
-
```
Comment (whipermr5) [inline src/main/webapp/partials/instructorHelpSessions.jsp]: Revert this unnecessary removal of a blank line.

File: src/main/webapp/partials/instructorHelpSessions.jsp [hunk 8]
```diff
@@ -566,10 +566,7 @@
             </form>
             <br>
             <br>
-          </div>
```
Comment (whipermr5) [inline src/main/webapp/partials/instructorHelpSessions.jsp]: It is line 519 that should be deleted instead. The problem was introduced [here](https://github.com/TEAMMATES/teammates/pull/8494/files#diff-2e74ce40cd9ed6f73d86dc9bc60dab4e) (but I failed to catch it then 😅).

File: src/main/webapp/partials/instructorHelpSessions.jsp (hunk 7 above)
Comment (Bzdeco) [reply src/main/webapp/partials/instructorHelpSessions.jsp]: Done

File: src/main/webapp/partials/instructorHelpSessions.jsp (hunk 8 above)
Comment (Bzdeco) [reply src/main/webapp/partials/instructorHelpSessions.jsp]: Done :smile:
